### ISSUE TITLE
Improve dependency management and considerations

### ISSUE THREAD

**Issue by jrobber** [GetStream/stream-chat-react-native#1462]
**Motivation**
I'm tired of wasting time trying to work around stream-chat library dependency issues.
I view stream chat as a library of convenience.  They are currently my highest time investment library just to keep my app working and building.
My two examples from just today.
1) Outdated dependencies break EAS builds if you use pnpm on your project, or a newer version of yarn.
This is not a minor issue.  stream-chat is now expo major 4 versions behind (I'm 1 behind on 44).  This is a build breaker!  That means this has been out dated for 1-2 years!
```
.
├─┬ stream-chat-expo
│ ├── ✕ unmet peer @react-native-community/netinfo@^6.0.0: found 7.1.3
│ ├── ✕ unmet peer expo@^41.0.1: found 44.0.6
│ ├── ✕ unmet peer expo-document-picker@^9.1.2: found 10.1.3
│ ├── ✕ unmet peer expo-file-system@^11.0.2: found 13.1.4
│ ├── ✕ unmet peer expo-haptics@^10.0.0: found 11.1.1
│ ├── ✕ unmet peer expo-image-manipulator@^9.1.0: found 10.2.1
│ ├── ✕ unmet peer expo-image-picker@^10.1.4: found 12.0.2
│ ├── ✕ unmet peer expo-media-library@^12.0.2: found 14.0.1
│ ├── ✕ unmet peer expo-sharing@^9.1.2: found 10.1.1
│ └─┬ stream-chat-react-native-core
│   └─┬ react-art
│     └── ✕ unmet peer react@17.0.2: found 17.0.1
```

2) Video chat added in 4.7.0 requires extra setup, and forces yet another dependency on me.

I don't have time to go through all those steps, and I don't want the overhead in my bundle.  
I had to downgrade to 4.6.0 to get a working build again, and now I'm locked into this version until I have a business use-case to accept the extra crap you guys just pushed onto me that I don't want/need in my app.

strict-peer-dependencies is a thing.  It's coming.  It's one of the only ways we'll get sane Javascript builds. Please get on board.

**Proposed solution**
1) Update the feature and release planning process with the following questions:
A) Does this release add any new dependencies that require any additional setup by the user.  If so, push that feature to another library, or at least a major version.
B) Does this release add any new dependencies that conflict with expo's version mapping of dependencies (stream-chat-expo only, probably)? If so, can you eliminate the need for those dependencies? If not, then don't use them until you can line up with expo's version mapping.
C) Does this release add any new dependencies that conflict with react & react-native's versions & features
IE - This is not you guys, but react-native themselves uses a use-subscription version that conflicts with their own react dependency version.
```
├─┬ react-native
│ └─┬ use-subscription
│   └── ✕ unmet peer react@^18.0.0: found 17.0.1
```

2) Implement better end-user testing that simulates the problems your consumers are facing.
Make some simple test apps that use your library and be sure it builds on react-native, and on eas.
Get one each running using npm, yarn & pnpm

**Acceptance Criteria**

3) Video is removed from stream-chat library and placed in it's on module so users can opt-in when they want.
4) Peer dependencies are updated ASAP to accurate version numbers
5) Improve the build pipeline to cover the two highly popular build systems and the 3 most popular package managers to ensure you aren't breaking builds.

** PS **
I'm sorry if this at all sounds a little snippy, I tried to reign it in.  But I've got a lot of things to get done in my app and I've lost so much time to what feels like just a lack of care/concern/awareness by the maintainers of this library.

** PSS **
Disagree if you want, but again, my opinion is that stream-chat is a service of convenience.  The more I keep having to fight stream just to get my builds working, the more likely I am to just do it my self where I have full control to do things right so I can focus on moving forward, not monkey patching and fighting conflicting work arounds.

**Comment by vishalnarkhede** [comment]
@jrobber thanks for detailed description about issues and we are currently trying our best to resolve them.

> Video is removed from stream-chat library and placed in it's on module so users can opt-in when they want.

Actually to tackle the exact issue described by you, we have made Video as an optional dependency. So core functionality of Chat SDK still works as it is, and you can choose to or not to add `react-native-video` or `expo-av` dependency on app. Are you facing any issues if you decide to skip installation of Video library?

**Comment by vanGalilea** [comment]
Hi @jrobber, thanks for reaching out. I want to address some of the topics you've mentioned and add to what @vishalnarkhede said.

I've created several issues that will cover some of your suggestions. feel free to follow their state here:
- https://github.com/GetStream/stream-chat-react-native/issues/1478
- https://github.com/GetStream/stream-chat-react-native/issues/1335
- Video related- installing video is optional, as @vishalnarkhede said. Opting in is already a possibility.

The rest of the points have been added to our internal backlog so that more updates will follow.

We're constantly doing our best to improve our SDK and keep its dependencies up to date. Fighting with the SDK should never be the case, and we would like to thank you for your feedback.

I will be closing this issue as we split it into several ones.
